src/cbuild/core/scanelf.py

```diff
@@ -239,7 +239,7 @@ def _scan_one(fpath):
     inf.close()
 
     # sanitize
-    if soname and len(soname) == 0:
+    if soname is not None and len(soname) == 0:
```

Comment:
the change should be correct as long as soname is either `None` or a string, if that's not true and i missed something my bad